engine/compiler/compiler.go

```diff
@@ -802,12 +807,13 @@ func packNetrcSecrets(spec *engine.Spec, netrc *drone.Netrc) bool {
 }
 
 // setNetrcSecretsToStep is a helper function that sets netrc secrets to a engine.Step
-func setNetrcSecretsToStep(step *engine.Step, spec *engine.Spec) {
+func setNetrcSecretsToStep(step *engine.Step, spec *engine.Spec, secretMap map[string]*engine.Secret) {
```

Comment:
this function is pretty simple, don't add secret map here - it makes it harder to follow

Reply:
I meant add to the secret map in the main function